jobclient/src/main/java/com/twosigma/cook/jobclient/JobClient.java

```diff
@@ -794,9 +822,38 @@ private void submit(List<Job> jobs, JobListener listener, String impersonatedUse
         if (null != statusLine && statusLine.getStatusCode() == HttpStatus.SC_CREATED) {
             isSuccess = true;
             _log.info("Successfully execute POST request with data " + json + " via uri " + _jobURI);
+        } else if (null != statusLine && statusLine.getStatusCode() >= HttpStatus.SC_INTERNAL_SERVER_ERROR) {
+            final String transactionTimeoutMessage = ":db.error/transaction-timeout";
```

Comment:
This seems like over-coupling ... can we make it so that the client doesn't have to know about the specific error coming from datomic in order to determine that there was a transaction timeout?

Reply:
Unfortunately this is the response string we get back from Cook scheduler: `:db.error/transaction-timeout Transaction timed out.`

Decoupling would require changes to the scheduler, probably return some sort of timeout error code, maybe `504 Gateway Timeout`?

Reply:
I think so, what are your thoughts?

Reply:
I wanted to avoid having to change the scheduler to limit this PR to the job-client, but I can make a separate PR for that change which goes in before this one.